objects/obj_popup/Mouse_50.gml

```diff
@@ -733,7 +733,7 @@ if (mouse_x>=xx+1465) and (mouse_y>=yy+499) and (mouse_x<xx+1576) and (mouse_y<y
         if (manag>10) then manag=0;
 
         for(i=0;i<501;i++){
-            if (obj_ini.name[target_[comp][i]]=="" and obj_ini.name[target_[comp][i]+1]=="") {
+            if (obj_ini.name[target_comp][i]=="" and obj_ini.name[target_comp][i+1]=="") {
```

Comment:
I think this change causes the crash

Reply:
ok, I probably didn't click Clean in GM after switching branches, it works now, sorry for the confusion, however, I spotted another bug, you can promote scouts to scouts, who then lose weapons:

![image](https://github.com/ZAB909/ChapterMaster/assets/63882831/9c3d6df5-71af-420b-8508-b75a1ff37262)